ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java

```diff
@@ -114,18 +127,22 @@ class RefreshIngestionAuthTokenTask extends TimerTask {
             public void run() {
                 try {
                     refreshIngestionAuthToken();
-                    timer.schedule(new RefreshIngestionAuthTokenTask(), defaultRefreshTime);
                 } catch (Exception e) {
                     log.error("Error in refreshIngestionAuthToken. " + e.getMessage(), e);
                     if (timer != null) {
-                        timer.schedule(new RefreshIngestionAuthTokenTask(), refreshTimeOnFailure);
+                        RefreshIngestionAuthTokenTask newRefreshIngestionAuthTokenTask = new RefreshIngestionAuthTokenTask();
+                        timer.schedule(newRefreshIngestionAuthTokenTask, refreshTimeOnFailure, defaultRefreshTime);
+                        refreshIngestionAuthTokenTask.cancel();
+                        refreshIngestionAuthTokenTask = newRefreshIngestionAuthTokenTask;
                     }
                 }
             }
         }
 
-        timer.schedule(new RefreshIngestionAuthTokenTask(), 0);
-        timer.schedule(new RefreshIngestionResourcesTask(), 0);
+        refreshIngestionAuthTokenTask = new RefreshIngestionAuthTokenTask();
```

Comment:
Really no need for that - task cant run after purge or cancel

What fixed it for you was just the closing of the client

Reply:
You're right, thanks!

In this change, I wanted to reuse the task objects instead of recreating them every execution, particularly because I was concerned about a potential memory leak. But I see there isn't one, and GC will handle each of the tasks.

I reverted those changes.